eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -707,22 +798,30 @@ fixupFees tx nativeScriptKeyWits = do
             & bodyTxL . feeTxBodyL .~ (fee <> changeTxOut ^. coinTxOutL)
   pure txWithFee
 
-fixupTx ::
+shelleyFixupTx ::
+  forall era.
   ShelleyEraImp era =>
   Tx era ->
   ImpTestM era (Tx era)
-fixupTx txRaw = do
-  (txWithScriptWits, nativeScriptKeyPairs) <- addScriptTxWits txRaw
-  txWithFee <- fixupFees txWithScriptWits (Map.keysSet nativeScriptKeyPairs)
-  txFixed <- addAddrTxWits txWithFee (Map.elems nativeScriptKeyPairs)
+shelleyFixupTx =
+  addNativeScriptTxWits
+    >=> addRootTxIn
+    -- Add the TxWits so we can calculate the fee correctly
+    >=> updateAddrTxWits
```

Comment:
This is step is redundant, fee calculation function `calcMinFeeTxNativeScriptWits` will already do that.
```suggestion
```